src/anbox/qemu/adb_message_processor.cpp

```diff
@@ -74,10 +74,11 @@ void AdbMessageProcessor::advance_state() {
 
       if (state_ == closed_by_host) {
         host_connector_.reset();
-        return;
+      } else {
+        wait_for_host_connection();
       }
 
-      wait_for_host_connection();
+      lock_.unlock();
```

Comment:
What is this actually fixing? That lock_ is kept lock is intentional.

Reply:
I found that sometimes the process will be blocked here, so it cannot exit normally.